libraries/node-core-library/src/Path.ts

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+
+/**
+ * Helper functions for working with file paths.
+ * @public
+ */
+export class Path {
+  /**
+   * Returns true if childPath refers to a location under parentPath.
+   * @remarks
+   * If the paths are relative, they will first be resolved using path.resolve().
+   */
+  public static isUnder(childPath: string, parentPath: string): boolean {
+    const relativePath: string = path.relative(path.resolve(childPath), path.resolve(parentPath));
+    return /^[.\/\\]+$/.test(relativePath);
+  }
```

Comment:
We are currently using this in this code:


[https://github.com/Microsoft/web-build-tools/blob/master/apps/rush-lib/src/cli/logic/PackageChangeAnalyzer.ts#L150](https://github.com/Microsoft/web-build-tools/blob/master/apps/rush-lib/src/cli/logic/PackageChangeAnalyzer.ts#L150)

Reply:
Fixed

---
In reply to: [169520850](https://github.com/Microsoft/web-build-tools/pull/537#discussion_r169520850) [](ancestors = 169520850)